src/buffer.rs

```diff
@@ -42,13 +44,10 @@ pub trait Pixel: Copy + Clone {
         Self::COLOR_MODEL
     }
 
-    /// ColorType for this pixel format
-    const COLOR_TYPE: ColorType;
+    /// Channels for this pixel format
+    const CHANNELS_TYPE: ChannelsType;
```

Comment:
`ColorType` has a method mapping to the `ChannelsType`. I would have assumed to instead gain a new:

```
fn channels_type() -> ChannelsType {
    ExtendedColorType::from(Self::COLOR_TYPE).channels_type()
}
```

Reply:
The problem is each color/pixel struct doesn't have an unequivocal `ColorType` that can be an associated const, but it does have an unequivocal `ChannelsType`, e.g., both `ColorType::Rgb8` and `ColorType::Rgb16` are color types of the `Rgb` struct. @fintelia's comment mentions this, although I like the genericness of the current struct design and went down the "pixel type per color type" path first before finding it a mess and retreating to this approach.

Generic associated items could make this easier, but those still seem a ways off.

Reply:
You're both right, the impl is too generic over `<T: Primitive>`.